torch/csrc/utils/tensor_new.cpp

```diff
@@ -262,7 +262,7 @@ Tensor internal_new_from_data(
     TORCH_CHECK(!pin_memory, "Can't pin tensor constructed from __cuda_array_interface__");
     auto tensor = tensor_from_cuda_array_interface(data);
     const auto& inferred_scalar_type = type_inference ? tensor.scalar_type() : scalar_type;
-    auto device = device_opt.has_value() ? *device_opt : at::Device(computeDeviceType(dispatch_key));
+    auto device = device_opt.has_value() ? *device_opt : at::Device(computeDeviceType(dispatch_key.first));
```

Comment:
Why in this case and the following case does the call not also include the index?

Reply:
Because this is the case when the tensor comes from `numpy`, so, the index is not needed.